packages/frontend/src/utils/getIsTxFinalized.ts

```diff
@@ -0,0 +1,20 @@
+import { orusThatRelyOnL1ConfirmationsForFinality } from 'src/config'
+import { getProviderByNetworkName } from 'src/utils/getProvider'
+import { getNetworkWaitConfirmations } from 'src/utils/networks'
+
+export async function getIsTxFinalized (
+  txBlockNumber: number | undefined,
+  chainSlug: string
+) {
+  if (!txBlockNumber) return false
+
+  const provider = getProviderByNetworkName(chainSlug)
+  if (orusThatRelyOnL1ConfirmationsForFinality.includes(chainSlug)) {
+    const finalizedBlock = await provider.getBlock('safe')
```

Comment:
On Optimism, this is intentional since it won't be exposed until Bedrock. It does work on the Goerli Optimism endpoint.

You are correct about Arbitrum. It appears as though Infura exposes this but Alchemy does not 🤔 . Will figure something out here, good call 👍